torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py

```diff
@@ -130,9 +130,25 @@ def aot_dispatch_base_graph(
             mod_when_exporting_non_strict, assigned_buffers
         )
 
-    saved_updated_flat_args_subclasses_desugared = pytree.tree_map_only(
-        torch.Tensor, lambda t: t.detach(), updated_flat_args_subclasses_desugared
-    )
+    # TODO: Refactor the following code so detach() persists item_memo
```

Comment:
It's hard for me to estimate how deep or shallow this rabbit hole is. If reviewers have a more accurate estimate and can push me one way or another (include the refactor in this PR or not), do let me know!

Reply:
You can do it on its own (and I'd recommend it), but basically if the detach implementation has some part that is in Python you're golden, and if it's entirely C++ give up and do something else lol